test/parallel/test-repl-recoverable.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const util = require('util');
+const repl = require('repl');
+
+var evalCount = 0;
+var recovered = false;
+var rendered = false;
+
+// A stream to push an array into a REPL
+function ArrayStream() {}
```

Comment:
`ArrayStream` is defined in `../common.js` now. Can you use that and just overwrite the `write()` function.